### PR TITLE
user_model_structure

### PR DESCRIPTION
**Pull request by thaisfreires**
AIMAID-21-Define-User-Model-and-Inheritance-Structure

### CODE REVIEW

File: app/src/main/java/util/userRole.kt
Comment (Pris-c) [inline app/src/main/java/util/userRole.kt:1]: The correct path is:
app/util/enum/UserRole

Also, just a reminder, the class name should start with an uppercase letter.

File: app/src/main/java/util/docType.kt
Comment (Pris-c) [inline app/src/main/java/util/docType.kt:1]: The correct path is:
app/util/enum/DocType

Also, just a reminder, the class name should start with an uppercase letter.

File: app/src/main/java/com/example/aima_id_app/data/model/submodel/DocUser.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * @property docType The type of the document (e.g., Passport, ID Card).
+ * @property docId The unique identifier of the document.
+ *
+ */
+class DocUser (val docType: docType, val docId: String) {
+
+    /**
+     * Displays the details of the user's document.
+     */
+    fun displayDetails(){
```
Comment (Pris-c) [inline app/src/main/java/com/example/aima_id_app/data/model/submodel/DocUser.kt:17]: I think we might not need this method.

File: app/src/main/java/com/example/aima_id_app/data/model/submodel/Address.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * @property street represents the name of the street of the user.
+ * @property number represents the number of the user's residence.
+ * @property city represents the city of the user's residence.
+ * @property postalCode represents the postal code of the given address.
+ */
+class Address (
+    val street: String,
+    val number: Int,
+    val city: String,
+    val postalCode: Int,
```
Comment (Pris-c) [inline app/src/main/java/com/example/aima_id_app/data/model/submodel/Address.kt]: The postalCode should be of type String.

File: app/src/main/java/com/example/aima_id_app/data/model/db_model/User.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+     val role: userRole,
+     val email: String,
+     val nif: String,
+     val name: String,
+     val dateOfBirth: Date,
+){
+     /**
+      * Displays the details of the user, including name, role, email, nif and date of birth.
+      */
+     open fun displayInfo(){
```
Comment (Pris-c) [inline app/src/main/java/com/example/aima_id_app/data/model/db_model/User.kt:25]: I think we might not need this method.

File: app/src/main/java/com/example/aima_id_app/data/model/db_model/AdminUser.kt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * @property name The name of the admin user.
+ * @property dateOfBirth The date of birth of the admin user.
+ */
+class AdminUser (role: userRole, email: String, nif: String, name: String, dateOfBirth: Date) :
+    User (userRole.ADMIN, email, nif , name, dateOfBirth)
+{
+    /**
+     * Displays the details of the admin user, including name, email, nif and date of birth.
+     */
+    override fun displayInfo() {
```
Comment (Pris-c) [inline app/src/main/java/com/example/aima_id_app/data/model/db_model/AdminUser.kt:21]: I think we might not need this method.

**Review by Pris-c — CHANGES_REQUESTED** [review]
Just a few small changes are needed.
I've added the comments directly in the code.

File: app/src/main/java/com/example/aima_id_app/data/model/submodel/Address.kt (hunk 2 above)
Comment (thaisfreires) [reply app/src/main/java/com/example/aima_id_app/data/model/submodel/Address.kt]: done

**Review by Pris-c — APPROVED** [review]

**Pull request merged**